tests/test_zipapp.py

```diff
@@ -0,0 +1,48 @@
+import os.path
+import subprocess
+import sys
+
+import pytest
+
+import virtualenv
+
+HERE = os.path.dirname(os.path.dirname(__file__))
+
+
+def _python(v):
+    return virtualenv.get_installed_pythons().get(v, "python{}".format(v))
+
+
+def _major_version(venv):
+    _, _, _, bin_dir = virtualenv.path_locations(venv)
+    exe = os.path.join(bin_dir, "python")
+    cmd = (exe, "-c", "import sys; print(sys.version_info[0])")
+    out = subprocess.check_output(cmd)
+    return int(out)
+
+
+@pytest.fixture(scope="session")
+def zipapp(tmp_path_factory):
+    if sys.version_info[:2] == (3, 4):
+        pytest.skip("zipapp was introduced in python3.5")
+    pyz = str(tmp_path_factory.mktemp(basename="zipapp") / "virtualenv.pyz")
+    subprocess.check_call(
+        (_python("3"), os.path.join(HERE, "tasks/make_zipapp.py"), "--root", virtualenv.HERE, "--dest", pyz)
+    )
+    return pyz
+
+
+def test_basic_invocation(zipapp, tmp_path):
+    venv = tmp_path / "venv"
+    subprocess.check_call((sys.executable, zipapp, str(venv)))
+    assert venv.exists()
+    assert _major_version(str(venv)) == sys.version_info[0]
+
+
+def test_invocation_dash_p(zipapp, tmp_path):
+    venv = tmp_path / "venv"
+    python = {2: _python("3"), 3: _python("2.7")}[sys.version_info[0]]
+    subprocess.check_call((sys.executable, zipapp, str(venv), "-p", python))
+    assert venv.exists()
```

Comment:
This would pass if all virtualenv would do is create this folder and then exit 🤔

Reply:
fortunately the rest of the test suite makes sure that isn't the case

Reply:
it makes sure in case of the source invocation though, not the zipapp interface, not? would we zipapp a single mkdir command I think think this would still pass.

Reply:
eh, talk about a fail 😆 , probably I should fix it :-) I would like to validate a simple call with the created virtualenv ```python -c 'import sys; print(sys.executable)'``` should return the same thing we calculated with the virtualenv path.

Reply:
since I'm already better than the status quo I'm going to drop this, feel free to follow up with your more strict requirements but I don't reasonably think you should hold me to higher standards than the status quo which I'm already exceeding